SYCL/DeviceLib/imf_half_type_cast.cpp

```diff
@@ -0,0 +1,589 @@
+        0,
+        0,
+        100,
+        101,
+        100,
+        std::numeric_limits<long long>::max(),
+        std::numeric_limits<long long>::min(),
+        0,
+        65504,
+        -65504,
+        -64960,
+        -100,
+        10008,
+        -19};
+
+    test_host(input_vals, ref_vals_rd, F_Half1(__imf_half2ll_rd));
+    test(device_queue, input_vals, ref_vals_rd, F_Half1(__imf_half2ll_rd));
+
+    test_host(input_vals, ref_vals_rn, F_Half1(__imf_half2ll_rn));
+    test(device_queue, input_vals, ref_vals_rn, F_Half1(__imf_half2ll_rn));
+
+    test_host(input_vals, ref_vals_ru, F_Half1(__imf_half2ll_ru));
+    test(device_queue, input_vals, ref_vals_ru, F_Half1(__imf_half2ll_ru));
+
+    test_host(input_vals, ref_vals_rz, F_Half1(__imf_half2ll_rz));
+    test(device_queue, input_vals, ref_vals_rz, F_Half1(__imf_half2ll_rz));
+  }
+
+  {
+    std::initializer_list<uint16_t> input_vals = {
+        0x0,
+        0x0256, // subnormal half-precision value
+        0x5648, // 100.5
+        0x5658, // 101.5
+        0x564C, // 100.75
+        0x7C00, // +infinity
+        0x7E00, // NAN
+        0x7BFF, // maximum half-precision value
+        0x73DF, // 16120
+        0x7AEE, // 56768
+        0xD648, // -100.5
+        0x70E3, // 10008
+        0x4CE3, // 19.546875
+    };
+
+    std::initializer_list<unsigned long long> ref_vals_rd = {
+        0,     0,     100,
+        101,   100,   std::numeric_limits<unsigned long long>::max(),
+        0,     65504, 16120,
+        56768, 0,     10008,
+        19};
+    std::initializer_list<unsigned long long> ref_vals_rn = {
+        0,     0,     100,
+        102,   101,   std::numeric_limits<unsigned long long>::max(),
+        0,     65504, 16120,
+        56768, 0,     10008,
+        20};
+    std::initializer_list<unsigned long long> ref_vals_ru = {
+        0,     1,     101,
+        102,   101,   std::numeric_limits<unsigned long long>::max(),
+        0,     65504, 16120,
+        56768, 0,     10008,
+        20};
+    std::initializer_list<unsigned long long> ref_vals_rz = {
+        0,     0,     100,
+        101,   100,   std::numeric_limits<unsigned long long>::max(),
+        0,     65504, 16120,
+        56768, 0,     10008,
+        19};
+
+    test_host(input_vals, ref_vals_rd, F_Half1(__imf_half2ull_rd));
+    test(device_queue, input_vals, ref_vals_rd, F_Half1(__imf_half2ull_rd));
+
+    test_host(input_vals, ref_vals_rn, F_Half1(__imf_half2ull_rn));
+    test(device_queue, input_vals, ref_vals_rn, F_Half1(__imf_half2ull_rn));
+
+    test_host(input_vals, ref_vals_ru, F_Half1(__imf_half2ull_ru));
+    test(device_queue, input_vals, ref_vals_ru, F_Half1(__imf_half2ull_ru));
+
+    test_host(input_vals, ref_vals_rz, F_Half1(__imf_half2ull_rz));
+    test(device_queue, input_vals, ref_vals_rz, F_Half1(__imf_half2ull_rz));
+  }
+
+  {
+    std::initializer_list<short> input_vals = {
+        0, 7, 49, -323, 2999, -32768, 32767, 32111, -11111, 16383};
```

Comment:
General question: do you check integer indefinite (-1.#IND) inputs somewhere?

Reply:
Hi, @akolesov-intel 
What do you mean by "integer indefinite"?

Thanks very much.

Reply:
Sorry, it is not related to conversions actually, I was wrong. I was thinking about exceptions handling